deps/gmp.mk

```diff
@@ -23,6 +23,10 @@ checksum-gmp: $(SRCCACHE)/gmp-$(GMP_VER).tar.bz2
 	$(JLCHECKSUM) $<
 
 $(SRCCACHE)/gmp-$(GMP_VER)/build-patched: $(SRCCACHE)/gmp-$(GMP_VER)/source-extracted
+	# Apply fix to avoid using Apple ARM reserved register X18
+	# Necessary for version 6.2.1, remove after next gmp release
+	cd $(dir $@) && patch -p1 < $(SRCDIR)/patches/gmp-HG-changeset.patch
```

Comment:
@vtjnash I didn't realize the GMP steps here didn't have the typical ladder of makefile dependencies on the patches.  Do we want to reorganize to have that?

Reply:
Do you have an example? Like in Yggdrasil's gmp?

Reply:
E.g. here: https://github.com/JuliaPackaging/Yggdrasil/blob/8d6c4cde2639a94eeee2c2e597fcc17603ebe7e2/G/GMP/common.jl#L25 ?

Reply:
In most of our `deps/*.mk` files, if we have a series of patches, we make a chain of makefile dependencies.  Libgit2 is a good example: https://github.com/JuliaLang/julia/blob/master/deps/libgit2.mk#L38-L58

Without that chain of dependencies, if we merge this, if someone has a checkout of Julia already on-disk, and they have already created the `build-patched` file, this patch will silently not be applied, because the `build-patched` file already exists.

Reply:
Thanks for the example. Pushed an update. :)